### PR TITLE
bug fix in member_config for having _id > 255

### PR DESCRIPTION
**Pull request by yuhanz**
I encountered this problem of having _id > 255 where mongodb considers it as invalid _id. The current recipe will get max_id += 1, which leads _id to grows to the top value.

The validation logic is here:
https://github.com/mongodb/mongo/blob/master/src/mongo/db/repl/member_config.cpp#L225

To fix the problem, instead of using max_id +=1, we pick a value between [0,255] for _id, excluding the _id's have existed in the member configs.

### CODE REVIEW

File: libraries/mongodb.rb [hunk 1]
```diff
@@ -166,10 +166,11 @@ def self.configure_replicaset(node, name, members)
           host = m['host']
           { '_id' => m['_id'], 'host' => host }.merge(rs_options[host])
         end
+        ids = (0...256).to_a - old_ids
         members_add = rs_members - old_members
         members_add.each do |m|
-          max_id += 1
```
Comment (thomas-riccardi) [inline libraries/mongodb.rb:171]: just use `max_id = (max_id + 1) % 256`

**Comment by thomas-riccardi** [comment]
This project is dead, the active fork is at https://github.com/chef-brigade/mongodb-cookbook where you can get this PR merged.

**Comment by thomas-riccardi** [comment]
Also, please use a stable url for the link to the mongodb server code in your commit message: https://github.com/mongodb/mongo/blob/r3.2.0/src/mongo/db/repl/member_config.cpp#L225

File: libraries/mongodb.rb (hunk 1 above)
Comment (yuhanz) [inline libraries/mongodb.rb:171]: I think %256 would risk id collision with old members

**Comment by yuhanz** [comment]
Thanks. 

I copied this patch to chef-brigade/mongodb-cookbook and created another pull request at: 
https://github.com/chef-brigade/mongodb-cookbook/pull/101

File: libraries/mongodb.rb (hunk 1 above)
Comment (thomas-riccardi) [inline libraries/mongodb.rb:171]: Indeed you are right!

**Pull request closed without merging**